integration-tools/src/main/java/uk/ac/bbsrc/tgac/miso/integration/dp5mirage/DP5MirageScan.java

```diff
@@ -0,0 +1,240 @@
+   */
+  @Override
+  public int getMaximumTubeCount() {
+    return scan.get("tubeBarcode").size();
+  }
+
+  /**
+   * @return the number of tubes currently in the box. This count may include tubes with barcodes that were successfully read, as well
+   * as tubes with barcodes that the scanner failed to read. See {@link #hasReadErrors()}
+   */
+  @Override
+  public int getTubeCount() {
+    int tubeCounter = 0;
+    for(DP5MirageScanPosition position: getFormattedScan()) {
+      if(!position.getDecodeStatus().equals("EMPTY")) {
+        tubeCounter++;
+      }
+    }
+    return tubeCounter;
+  }
+
+  /**
+   * @return true if the scanner failed to read the barcode on any tube(s) in the box; false otherwise. An empty position (no tube) does
+   * not count as a read error; there must actually be a tube in the position to have a read error
+   */
+  @Override
+  public boolean hasReadErrors() {
+    for(DP5MirageScanPosition position: getFormattedScan()) {
+      if(position.getDecodeStatus().equals("ERROR")) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  /**
+   * @return a list of each position where there is a tube, but the scanner failed to read its barcode. Positions are represented by
+   * row letter and two-digit column number (e.g. "A01" for top-left)
+   */
+  @Override
+  public List<String> getReadErrorPositions() {
+    final String[][] barcodes = getBarcodesArray();
+    final String noRead = getNoReadLabel();
+    List<String> positions = new ArrayList<>();
+
+    for (int row = 0; row < barcodes.length; row++) {
+      for (int col = 0; col < barcodes[row].length; col++) {
+        if (noRead.equals(barcodes[row][col])) positions.add(BoxUtils.getPositionString(row, col));
+      }
+    }
+    return positions;
+  }
+
+  /**
+   * @return the number of rows in the scanned box
+   */
+  @Override
+  public int getRowCount() {
+    return scan.get("tubeBarcode").get(scan.get("tubeBarcode").size() -1).get("row").asInt();
+  }
+
+  /**
+   * @return the number of columns in the scanned box
+   */
+  @Override
+  public int getColumnCount() {
+    return scan.get("tubeBarcode").get(scan.get("tubeBarcode").size() -1).get("column").asInt();
+  }
+
+  /**
+   * @return the String label for a single unreadable tube
+   */
+  @Override
+  public String getNoReadLabel() {
+    return "\"null\"";
+  }
+
+  /**
+   * @return the String label for the absence of a tube
+   */
+  @Override
+  public String getNoTubeLabel() {
+    return "EMPTY";
+  }
+
+  /**
+   * @return array of DP5MirageScanPosition objects
+   */
+  public DP5MirageScanPosition[] getFormattedScan() {
+    ObjectMapper objectMapper = new ObjectMapper();
+    List<DP5MirageScanPosition> positions = objectMapper.convertValue(scan.get("tubeBarcode"),
+        new TypeReference<>() {
+        });
+
+    return positions.toArray(new DP5MirageScanPosition[scan.get("tubeBarcode").size()]);
```

Comment:
Should be private since it's for use within this class only. I think this `DP5MirageScanPosition` class is probably unnecessary though. The only extra thing you're storing in it is the decode status and that should be represented in the barcodes map/collection already. `getNoReadLabel` and `getNoTubeLabel` are meant to be the values that are used in place of barcodes where there is an error or a missing tube